### DISCUSSION TITLE
Question about data storage

### DISCUSSION THREAD

**Discussion by glensimister** [freenet/freenet-core#308] [Q&A] [answered]
Hi folks. I just found out about Locutus. It looks interesting but i'm still trying to get my head around some of the concepts. How/where is data stored? I understand that it is a decentralized key/value store, but storing large files on a P2P isn't really scalable.

I read elsewhere that "Locutus is more like a cache than long-term storage, so data can fall out of the network if nobody is requesting it, but information publishers can serve as a backup for data stored in the network, replacing it if it falls out".

Can somebody explain this to me? Does it/can it use external storage adapters?

Cheers

**Comment by ple1n** [comment, marked as answer]
>  How/where is data stored?

Both seeders and a few logically close peers store a particular piece of data. 

> but storing large files on a P2P isn't really scalable.

Data stored on Locutus is usually not that large. For a video hosting site you may store the moderation metadata on Locutus and store videos elsewhere, like IPFS. Even if you store videos on Locutus, the actual storage is mostly done by seeders, which doesn't bloat the network.

> Does it/can it use external storage adapters?

Locutus protocol itself doesn't, but applications using it can use whatever is available. Locutus apps that run in the browsers can connect to metamask, request IPFS resources etc. A desktop Locutus app is the same as native apps, so, no limitations.
